netket/vqs/mc/mc_state/state.py

```diff
@@ -98,21 +98,26 @@ class MCState(VariationalState):
     """An Optional PyTree encoding a mutable state of the model that is not trained."""
 
     _sampler: Sampler
-    """The sampler used to sample the hilbert space."""
+    """The sampler used to sample the Hilbert space."""
     sampler_state: SamplerState
-    """The current state of the sampler"""
+    """The current state of the sampler."""
 
     _n_samples: int = 0
-    """Total number of samples across all mpi processes."""
+    """Total number of samples across all chains and MPI processes."""
+    _n_samples_per_rank: int = 0
+    """Number of samples across all chains on one MPI process."""
+    _chain_length: int = 0
```

Comment:
Why do we cache those things now?
We can compute them just by keeping 1 thing around, and I think it ensures there will be fewer problems..

Reply:
Ok, let me try not to cache them